crates/bevy_asset/src/io/android.rs

```diff
@@ -46,15 +45,53 @@ impl AssetReader for AndroidAssetReader {
 
     async fn read_directory<'a>(
         &'a self,
-        _path: &'a Path,
+        path: &'a Path,
     ) -> Result<Box<PathStream>, AssetReaderError> {
-        let stream: Box<PathStream> = Box::new(EmptyPathStream);
-        error!("Reading directories is not supported with the AndroidAssetReader");
-        Ok(stream)
+        let asset_manager = bevy_winit::ANDROID_APP
+            .get()
+            .expect("Bevy must be setup with the #[bevy_main] macro on Android")
+            .asset_manager();
+        let opened_assets_dir = asset_manager
+            .open_dir(&CString::new(path.to_str().unwrap()).unwrap())
+            .ok_or(AssetReaderError::NotFound(path.to_path_buf()))?;
+
+        let mapped_stream = opened_assets_dir
+            .filter_map(move |f| {
+                let file_path = path.join(Path::new(f.to_str().unwrap()));
+                // filter out meta files as they are not considered assets
+                if let Some(ext) = file_path.extension().and_then(|e| e.to_str()) {
```

Comment:
`.as_asset_file()` or something could probably be its own function since that seems useful to have generally speaking.

Reply:
I agree, but I think it might be better to just extract out the "meta" file detection and filtering. I am worried that something like `.as_asset_file()` might be confusing (since it would take a file path and then either return `Some` or `None`, depending on this condition - and that `Some` value would be the original value unchanged).